client/lib/user-settings/index.js

```diff
@@ -156,8 +173,8 @@ UserSettings.prototype.cancelPendingEmailChange = function( callback ) {
  */
 UserSettings.prototype.getOriginalSetting = function( settingName ) {
 	var setting = null;
-	if ( this.settings && this.settings[ settingName ] ) {
-		setting = this.settings[ settingName ];
+	if ( this.settings && get( this.settings, settingName ) ) {
```

Comment:
This may be outside the scope of this PR but I believe theres an issue here (before & after the change).
I think we could quite easily fix that here though, if you don't mind?

Here's a quick demo showing the issue... basically any falsey value would be replaced with `null`:

https://runkit.com/spen/599cd633888c48001253cc1d

I think something like so should do the trick:
```js
getOriginalSetting = settingName =>
	has( notThis.settings, settingName )
        ? get( notThis.settings, settingName )
        : null;
```

Reply:
Commenting for continuity: As per our discussion in Slack, we'll address this one outside the scope of this PR in order to isolate any potential side effects.

Reply:
Just a note on the code example I gave above - that can be squished down further:
```js
getOriginalSetting = settingName => get( this.settings, settingName, null );
```